src/directives/markers.js

```diff
@@ -269,6 +269,9 @@ angular.module('ui-leaflet').directive('markers',
                             return;
                         _create(newMarkers, oldMarkers);
                     });
+                    scope.$on('$destroy', function () {
+                        _destroy(leafletScope.markers, {}, leafletMarkers, map, layers);
```

Comment:
Normally this would make sense. But the markers are on the same scope as the map. So deleting the markers should already happen since the map is being removed.

https://github.com/angular-ui/ui-leaflet/blob/c30e5d1107aaf7c989751c181c4881f4c421d343/dist/ui-leaflet.js#L137